streaming/src/main/scala/org/apache/spark/streaming/rdd/WriteAheadLogBackedBlockRDD.scala

```diff
@@ -96,9 +99,27 @@ class WriteAheadLogBackedBlockRDD[T: ClassTag](
         logDebug(s"Read partition data of $this from block manager, block $blockId")
         iterator
       case None => // Data not found in Block Manager, grab it from write ahead log file
-        val reader = new WriteAheadLogRandomReader(partition.segment.path, hadoopConf)
-        val dataRead = reader.read(partition.segment)
-        reader.close()
+        var dataRead: ByteBuffer = null
+        var writeAheadLog: WriteAheadLog = null
+        try {
+          val dummyDirectory = FileUtils.getTempDirectoryPath()
+          writeAheadLog = WriteAheadLogUtils.createLogForReceiver(
+            SparkEnv.get.conf, dummyDirectory, hadoopConf)
```

Comment:
The log directory needs to be passed through the `WriteAheadLogUtils.createLogForXXX()`. If you want to hide it from this method, and pass it through the SparkConf, then every place where `WriteAheadLogUtils.createLogForXXX()` needs to be called, we need to add the following. 
```
val walConf = SparkEnv.get.conf.clone()
walConf.set("logdir", ....)
```

IMO that duplicates code everywhere and uglier that this dummy dir approach.